### PR TITLE
CLOUDP-69434: Milestone: Agents

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-69434

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

**List MongoDB Agent versions in the provided project.**

```
./bin/mongocli om agents versions ls -P ops
HOSTNAME    ADDRESS           VERSION        IS DEPRECATED
andreaMac   0:0:0:0:0:0:0:1   10.20.0.6609   false

```
```
./bin/mongocli om agents versions ls -P ops -o json
{
  "count": 1,
  "entries": [
    {
      "address": "0:0:0:0:0:0:0:1",
      "confCount": 2134,
      "hostname": "andreaMac",
      "hostnameShort": "andreaMac",
      "isManaged": true,
      "isModule": true,
      "isPrimary": false,
      "isVersionOld": false,
      "isVersionDeprecated": false,
      "lastConf": 1602666663325,
      "lastPing": 0,
      "msSinceLastConf": 9073,
      "numProcess": 1,
      "pingCount": 0,
      "pingState": "ok",
      "tagName": "",
      "version": "10.20.0.6609"
    }
  ],
  "isAnyAgentNotManaged": false,
  "isAnyAgentVersionDeprecated": false,
  "isAnyAgentVersionOld": false,
  "links": [
    {
      "rel": "self",
      "href": "http://localhost:8080/api/public/v1.0/groups/5f84269c7af9e2081998b599/agents/current"
    },
    {
      "rel": "http://mms.mongodb.com/group",
      "href": "http://localhost:8080/api/public/v1.0/groups/5f84269c7af9e2081998b599"
    }
  ],
  "minimumAgentVersionDetected": "10.20.0.6609",
  "minimumVersion": "10.15.0.6463"
}
```

**List earliest and latest versions of MongoDB Agent, MongoDB Tools, and BI Connector in your environment**

```
./bin/mongocli om softwareComponents ls -P ops
AUTOMATION VERSION   BI CONNECTOR VERSION   MONGODB TOOLS VERSION
10.20.0.6609         2.14.0                 100.1.0
```

```
./bin/mongocli om softwareComponents ls -P ops -o json
{
  "automationVersion": "10.20.0.6609",
  "automationMinimumVersion": "10.15.0.6463",
  "biConnectorVersion": "2.14.0",
  "biConnectorMinimumVersion": "2.3.1",
  "links": [
    {
      "rel": "self",
      "href": "http://localhost:8080/api/public/v1.0/softwareComponents/versions"
    }
  ],
  "mongoDbToolsVersion": "100.1.0"
}
```

### CODE REVIEW

File: internal/store/agents.go [hunk 1]
```diff
@@ -97,3 +105,25 @@ func (s *Store) DeleteAgentAPIKey(projectID, keyID string) error {
...
+// Agents encapsulates the logic to manage different cloud providers
+func (s *Store) AgentProjectVersions(projectID string) (*opsmngr.AgentVersions, error) {
+	switch s.service {
+	case config.OpsManagerService:
+		result, _, err := s.client.(*opsmngr.Client).Agents.ProjectVersions(context.Background(), projectID)
+		return result, err
+	default:
+		return nil, fmt.Errorf("unsupported service: %s", s.service)
+	}
+}
```
Comment (andreaangiolillo) [inline internal/store/agents.go:129]: I don't see these two endpoints for Cloud Manager https://docs.cloudmanager.mongodb.com/reference/api/agents/

Comment (gssbzn) [reply internal/store/agents.go:129]: AgentGlobalVersions is OM only but `AgentProjectVersions` should work with CM, if it's not documented let's leave like that

**Review by gssbzn — APPROVED** [review]
Couple of small nits but nothing blocking

**Comment by andreaangiolillo** [comment]
I am going to address these comments in the next PR

**Pull request merged**